### ISSUE TITLE
Additional information starts from "maxed up" pokemons, & Q about its range

### ISSUE THREAD

**Issue by rpgrca** [farkam135/GoIV#223]
This probably requires more feedback: I'm L27, so when I choose advanced information in any pokemon, it would show me the CP it would have if I pumped it all the way up to L28.5, even for a L20 bulbasaur. Wouldn't it be more useful to automatically show the data for the immediately superior evolution at the current pokemon's level instead?

![2016-08-27 13 29 36_](https://cloud.githubusercontent.com/assets/15602473/18027708/767f8eae-6c41-11e6-9b37-69bea44a2b1c.jpg)

This is based in a) there's no way I'd spend 49 candies and 60500 stardust pushing such bulbasaur til L28.5 without knowing its moveset will be worth unless it's already at its maximum evolution; and b) I'm more interested in knowing if I evolved this bulbasaur, will he get a higher CP amount than the ivysaur I already got to spend the candies?

Also, if I choose "ivysaur", the amount of candies should be updated with the 25 candies needed to go from bulbasaur to ivysaur (and if I were to choose venasaur, it should also contain the 100 candies needed to jump from ivysaur to venasaur). Right now, it gets stuck at 0 candies.

![2016-08-27 13 29 45_](https://cloud.githubusercontent.com/assets/15602473/18027709/768863d0-6c41-11e6-8ddc-650d56f5e481.jpg)

In other words, I think clicking on Additional information for this bulbasaur should select Ivysaur, L20 (current level) and 25 candies (cost to upgrade from bulbasaur to ivysaur).

**Comment by nahojjjen** [comment]
That's how the function should work, we just didn't implement it because we wanted to rush out the new UI.

**Comment by Blaisorblade** [comment]
> I'm L27, so when I choose advanced information in any pokemon, it would show me the CP it would have if I pumped it all the way up to L28.5, even for a L20 bulbasaur. Wouldn't it be more useful to automatically show the data for the immediately superior evolution at the current pokemon's level instead?

I agree everybody has different preferences and picking a default is hard.

Here's my rationale when I proposed this as an easy default: we wanted it to be easy to get both "current level (here 20)" and "max level you can reach now (here 28.5)". To get to the current level, you just go all the way to the left which is easy(-ish). If we default to 20, to get to 28.5 you'd have to go all the way to... the center? That's not harder to pick.

I'd prefer to have some button to "get to the current max (28.5)", but I'm not sure which control to add for a good UX—some of the UI/UX guys should look at this, maybe @oskomonzon has ideas?

**Comment by oskomonzon** [comment]
This one is a little tricky because everyone uses the app according to their needs.

Generally speaking, lot's of trainers want to know how much resources they need to pump up a Pokémon to the max they are able to reach. In that scenario starting with the slider at the maximum allowed makes perfect sense: As @Blaisorblade said, it's easier to scale it down shooting to the left from that state than starting at current Pokémon level and having to aim somewhere inbetween.

Yes, we could add an extra button for it to go straight to the current max and initialize the slider at 0. That will cater @rpgrca's proposed scenario. But for other users will imply doing an extra tap to get the information they require (as is the case now for those who want to start with the slider to the left), effectively moving the issue around while adding a little more clutter to the screen with an extra button.

That being said, I agree with the proposed change to factor evolution costs, I'd say it would be best if we manage to keep it clear enough for the trainer to know we are talking about Evolution cost + Power up cost. Maybe something like the CP comparison.

Overall, the UI/UX changes still need some passes but I think we are slowly getting there.
I'd like to hear @nahojjjen input on the matter.

**Comment by nahojjjen** [comment]
I want to show evolution + level candy cost if the user checks a prediction which would require both.

**Comment by rpgrca** [comment]
Another point to discuss: the current bar goes from the current Pokemon level to 40. Shouldn't it cap at the current player level + 1.5 instead, the maximum that can be achieved? What is the purpose of knowing the CP of a Pokemon at maximum level when you are probably months away from that point? For comparison purposes, comparing two pokemon at the current maximum level should be enough, and would allow to slap the bar completely to the left to check the current information, or completely to the right to see the maximum available at this moment.

**Comment by Blaisorblade** [comment]
> Another point to discuss: the current bar goes from the current Pokemon level to 40. Shouldn't it cap at the current player level + 1.5 instead, the maximum that can be achieved?

FWIW, I asked that already and @nahojjjen personally uses this info, so there's at least one user who wants this freedom. I confess I'm not sure what for, but I guess it's e.g. good to compare your Pokemons to ones of other players?

**Comment by nahojjjen** [comment]
If I'm the only user who cares then it'd be fine if it was removed, I'll get over it. I think I'm not the only one though.

**Comment by oskomonzon** [comment]
> FWIW, I asked that already and @nahojjjen personally uses this info, so there's at least one user who wants this freedom. I confess I'm not sure what for, but I guess it's e.g. good to compare your Pokemons to ones of other players?

That's the main reason I like the un-capped slider too. It's also useful to see how the Pokémon will escalate and how much resources I'll need in a few levels (Up 'til 30 one can easily lvl up more than once per week).

**Comment by rpgrca** [comment]
That's useful, though, so I'd suggest adding an option whether you want the cap to be the maximum "theorical" level or the maximum "achievable" level, along with whether to include level candy cost? I know you all hate having too many settings, though.

Using @Blaisorblade's rationale that going all the way left to get the current level is simpler, going all the way right to get the maximum achievable vale is much simpler than, after going all the way to the left to check the current value, trying to find where was the maximum current cap.

**Comment by Paxtez** [comment]
Maybe adding some flags on the slider might make it more clear what the slider means.
^Current       ^Level Max     ^Max

**Comment by Blaisorblade** [comment]
We've changed the default to be on the left.

> Wouldn't it be more useful to automatically show the data for the immediately superior evolution at the current pokemon's level instead?

Hm, Haven't done that yet, and I agree that's what I want more often. Should do.

**Comment by Blaisorblade** [comment]
With #427 we achieved the core of this request, that is:

> Wouldn't it be more useful to automatically show the data for the immediately superior evolution at the current pokemon's level instead?

This is in master adn will be in 3.2.0; for any issues with the compromise we have there, please open new issues, but please try it out first 😉
